### PR TITLE
Snippets for Jenkins, Page Object Pattern

### PR DESCRIPTION
**Pull request by bradumbaugh**
Snippets for: 1) Building / testing / deploying a Xamarin.iOS
application using Jenkins; 2) Page Object Pattern in C#

### CODE REVIEW

**Comment by brminnick** [comment]
@bradumbaugh What do you think about making `PageBase` an `abstract` class?

This way we can make `VerifyPresent()` an abstract method, and every class that inherits `PageBase` is forced to implement `VerifyPresent()`.

```
using System;
using NUnit.Framework;
using Xamarin.UITest;
using Xamarin.UITest.Queries;
using System.Linq;
using Xamarin.UITest.iOS;

namespace FormsKitchenSink.UITests.Pages
{
    public abstract class PageBase
    {
        #region properties & fields
        protected readonly IApp app;
        protected readonly Platform platform;
        #endregion

        #region methods
        public abstract void VerifyPresent();
        #endregion

        #region constructors
        protected PageBase(IApp app, Platform platform)
        {
            this.app = app;
            this.platform = platform;
        }    
        #endregion
    }
}
```

**Comment by michael-watson** [comment]
I don't like the ```VerifyPresent``` method. It's called so many times (in ```SwitchValue``` Get, ```ToggeSwitch``` method) that in Test Cloud it will add additional time to the tests. I think we should cut out ```VerifyPresent``` from those two places and it will be much cleaner. People will think that they need to call ```VerifyPresent``` in each method that interacts with UI and could get confused by seeing other parts where ```VerifyPresent``` isn't called in the method. So the question then becomes, *when do we call ```VerifyPresent```...*

@bradumbaugh would it work removing ```VerifyPresent``` from these spots? added as comments below

File: samples/Snippets/PageObjectPatternExample/SwitchPage1.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                        colorDescription == iOSWhite ? White : OtherColor;
+                }
+            }
+        }
+
+        public bool SwitchValue 
+        {
+            get
+            {
+                VerifyPresent();
```
Comment (michael-watson) [inline samples/Snippets/PageObjectPatternExample/SwitchPage1.cs]: Can we remove this?

File: samples/Snippets/PageObjectPatternExample/PageBase.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public class PageBase
+    {
+        #region properties & fields
+        protected readonly IApp app;
+        protected readonly Platform platform;
+        #endregion
+
+        #region methods
+        public virtual void VerifyPresent() { }
```
Comment (michael-watson) [inline samples/Snippets/PageObjectPatternExample/PageBase.cs]: I think we should add a note discussing why this is there and how it should be used. I don't know the best way but let's have an open discussion and figure something out

**Comment by bradumbaugh** [comment]
@michael-watson @brminnick I like all these ideas. I've implemented them and will submit a new pull request.

**Pull request closed without merging**